test/int/cluster_test.go

```diff
@@ -90,14 +109,10 @@ var _ = Describe("AtlasCluster", func() {
 			atlasCluster, _, err := atlasClient.Clusters.Get(context.Background(), createdProject.Status.ID, createdCluster.Spec.Name)
 			Expect(err).ToNot(HaveOccurred())
 
-			createdAtlasCluster, err := createdCluster.Spec.Cluster()
+			mergedCluster, err := atlascluster.MergedCluster(*atlasCluster, createdCluster.Spec)
```

Comment:
using the `MergedCluster` to compare clusters in total.
This could be error-prone in case if `MergedCluster` has bugs in it - that's why the unit tests above were extended